engines/scumm/script_v6.cpp

```diff
@@ -2422,6 +2422,20 @@ void ScummEngine_v6::o6_talkActor() {
 	_string[0].loadDefault();
 	actorTalk(_scriptPointer);
 
+	// WORKAROUND: Dr Fred's first reaction line about Hoagie's and Laverne's
+	// units after receiving a new diamond is unused because of missing
+	// wait.waitForMessage() calls. We always simulate this opcode when
+	// triggering Dr Fred's lines in this part of the script, since there is
+	// no stable offset for all the floppy, CD and translated versions, and
+	// no easy way to only target the impacted lines.
+	if (_game.id == GID_TENTACLE && vm.slot[_currentScript].number == 9
+		&& vm.localvar[_currentScript][0] == 216 && _actorToPrintStrFor == 4 && _enableEnhancements) {
```

Comment:
It would be good to also check for the offset, like in the Dig workaround below

Reply:
I don't believe I can: the offsets appear to change between the CD and floppy releases, the official translations, and probably in every fan-made translation :( 

The Dig in "luckier" in this regard, because the real text is in the `LANGUAGE.BND` translation table, the script just containing a static reference to it (e.g. `/STOP.008/...`) so the offset shouldn't change. DOTT still embedded the real text in its scripts, though.

So the `vm.localvar[_currentScript][0] == 216 && _actorToPrintStrFor == 4` conditions try to limit my workaround to a smaller part of the script, but indeed it's going to be applied to some Dr. Fred lines where it's unnecessary.

But, empirically, it looks like the existing `_forcedWaitForMessage` trick doesn't mind if a line already has its `wait.waitForMessage()`, so my full workaround appears to work and not cause any problem.

If someone sees a better solution, though…

Reply:
Thanks for clearing that out